tests/baselines/reference/transformApi/transformsCorrectly.transformAddImportStar.js

```diff
@@ -1,13 +1 @@
-System.register(["./comp1"], function (exports_1, context_1) {
-    var i0;
-    var __moduleName = context_1 && context_1.id;
-    return {
-        setters: [
-            function (i0_1) {
-                i0 = i0_1;
-            }
-        ],
-        execute: function () {
-        }
-    };
-});
+import * as i0 from "./comp1";

```

Comment:
`transpileModule` expected all inputs are to be considered modules because `isolatedModules` expected all inputs to be modules. I think the issue for this test is that the new `SourceFile` we produce in the transform doesn't have an `externalModuleIndicator`, since we don't set that in the factory method, though we probably should.

Reply:
> ... though we probably should.

This is a bit of a tricky problem. A file that _starts_ as a Module, but might have its import/export declarations transformed out should probably _stay_ a Module (since that's how we type-checked it, and how it was written). It's a bit more confusing when you have a file that may or may not have been a Module, but you add import/export syntax to it.

Right now, a Module stays a Module because we'll just copy the module indicator over verbatim. We don't really use that contents of that value in transforms, merely its truthiness. If we switch to updating that value, we have to take care that we don't set it to a falsy value when the file was a Module to begin with, otherwise we might break consumers of the API.

Reply:
I think most of these tests were written under the assumption that each file would be treated as a module by `transpileModule`. It may be fine to just amend the tests with an `export {}` to preserve this, since most of these tests weren't specifically testing that assumption, but some other mechanism instead.

Reply:
I think the behavior I’ve implemented in this PR is better in some sense, but I worry about breaking people who were using our transpile API, but never using us to type check. Should we be concerned about that?